### PR TITLE
Rework buttons for control

### PR DESCRIPTION
**Pull request by StimkaT**

### CODE REVIEW

File: projects/app/src/lib/services/messagesFromElectron.service.ts [hunk 1]
```diff
@@ -23,7 +23,9 @@ export class MessagesFromElectronService {
     } else if (message.event === 'USB_DEVICES_PORT_IS_CLOSED') {
       this.store.dispatch(setClosePort({ deviceName: message.data.deviceName }));
     } else if (message.event === 'FROM_USB_DEVICE') {
+      console.log('Ошибка', message)
       this.store.dispatch(messageFromDevice({ message }));
+
```
Comment (StimkaT) [inline projects/app/src/lib/services/messagesFromElectron.service.ts:28]: remove

File: projects/app/src/lib/services/messagesFromElectron.service.ts [hunk 2]
```diff
@@ -23,7 +23,9 @@ export class MessagesFromElectronService {
     } else if (message.event === 'USB_DEVICES_PORT_IS_CLOSED') {
       this.store.dispatch(setClosePort({ deviceName: message.data.deviceName }));
     } else if (message.event === 'FROM_USB_DEVICE') {
+      console.log('Ошибка', message)
```
Comment (StimkaT) [inline projects/app/src/lib/services/messagesFromElectron.service.ts:26]: remove

File: projects/tripod/arduino/widget-tripod/widget-tripod.ino [hunk 3]
```diff
@@ -114,21 +116,44 @@ void events(const String message) {
...
+        if (s1 != 0) {
+          info.s1 = s1;
+          // sendDeviceInfo(timestamp);
+        }
+        if (s2 != 0) {
+          info.s2 = s2;
+          // sendDeviceInfo(timestamp);
+        }
       }
+
```
Comment (StimkaT) [inline projects/tripod/arduino/widget-tripod/widget-tripod.ino:152]: remove

File: projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts [hunk 4]
```diff
@@ -59,50 +62,35 @@ export class ControlButtonsContainer {
...
-  //       this.buttonClick('LEFT');
-  //       break;
-  //     case 'ArrowRight':
-  //       this.buttonClick('RIGHT');
-  //       break;
-  //   }
-  // }
-}
+    else if ($event.event === 'ControlButtonsComponent:BUTTON_CLICKED' && $event.note === 'mouseup') {
+      clearTimeout(this.timerId); // Скидываем таймер
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts:79]: Remove comment

File: projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts [hunk 5]
```diff
@@ -59,50 +62,35 @@ export class ControlButtonsContainer {
...
+      if (!this.isLongPress) {
+        this.store.dispatch(sendDirection({
+          direction: $event.data,
+          m: 1
+        }));
+        console.log('Кликнул', $event)
```
Comment (StimkaT) [inline projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts:85]: remove all console.log -s

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 6]
```diff
@@ -1,20 +1,120 @@
...
+      console.log('ОТПУСТИЛИ r')
     }
+
+    else if (message.key === 'ArrowUp' && note === 'down' && !this.isArrowUpDown) {
+      this.isArrowUpDown = true;
+      this.store.dispatch(sendDirection({
+        direction: 'UP',
+        m: 2
+      }));
+      console.log('НАЖАЛИ u')
```
Comment (StimkaT) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:85]: remove all console.log -s

File: projects/ui/src/lib/components/control-buttons/control-buttons.component.ts [hunk 7]
```diff
@@ -12,12 +12,17 @@ import { MatIconModule } from '@angular/material/icon';
...
+
+  buttonHold(data: string, note: any) {
     const message = {
       event: 'ControlButtonsComponent:BUTTON_CLICKED',
       data,
+      note,
     };
     this.emitter.emit(message);
-    console.log(message)
+    // console.log('Отправил', note, data)
```
Comment (StimkaT) [inline projects/ui/src/lib/components/control-buttons/control-buttons.component.ts:25]: remove comment

File: projects/ui/src/lib/components/control-buttons/control-buttons.component.ts [hunk 8]
```diff
@@ -12,12 +12,17 @@ import { MatIconModule } from '@angular/material/icon';
 export class ControlButtonsComponent {
   @Output() emitter = new EventEmitter()
 
-  buttonClick(data: string) {
+
```
Comment (StimkaT) [inline projects/ui/src/lib/components/control-buttons/control-buttons.component.ts:15]: empty line

File: projects/tripod/arduino/widget-tripod/widget-tripod.ino [hunk 9]
```diff
@@ -114,21 +116,44 @@ void events(const String message) {
...
       }
-      if (s2 != 0) {
-        info.s2 += s2;
-        sendDeviceInfo(timestamp);
+
+      if (info.m == 2) { //mode 2
+        if (s1 != 0 && info.s1 == s1) {
+          info.s1 = 0;
+          // sendDeviceInfo(timestamp);
+          return;
```
Comment (sislex) [inline projects/tripod/arduino/widget-tripod/widget-tripod.ino:136]: remove return

File: projects/tripod/arduino/widget-tripod/widget-tripod.ino [hunk 10]
```diff
@@ -114,21 +116,44 @@ void events(const String message) {
...
+        }
+        if (s2 != 0 && info.s2 == s2) {
+          info.s2 = 0;
+          // sendDeviceInfo(timestamp);
+          return;
```
Comment (sislex) [inline projects/tripod/arduino/widget-tripod/widget-tripod.ino:141]: remove return

File: projects/tripod/src/lib/+state/skins/move-skin/view/messages.effects.ts [hunk 11]
```diff
@@ -54,32 +54,58 @@ export class SetButtonEffects {
...
-            };
-          } else if (direction === 'DOWN') {
-            data = {
-              s2: -steps,
-            };
+          if (m === 1) {
+            if (direction === 'RIGHT') {
+              data = {
+                s1: steps,
+              };
```
Comment (sislex) [inline projects/tripod/src/lib/+state/skins/move-skin/view/messages.effects.ts:64]: if (direction === 'RIGHT') {
              data = {
                s1: m === 1 ? steps : 1,
              };

File: projects/tripod/src/lib/+state/skins/move-skin/view/move-view-skin.actions.ts [hunk 12]
```diff
@@ -23,5 +23,16 @@ export const setActiveDelay = createAction(
 
 export const sendDirection = createAction(
   '[Skin] sendDirection',
-  props<{ direction: string }>()
-);
\ No newline at end of file
+  props<{ direction: string, m: any }>()
```
Comment (sislex) [inline projects/tripod/src/lib/+state/skins/move-skin/view/move-view-skin.actions.ts:26]: m: number

File: projects/tripod/src/lib/+state/skins/move-skin/view/move-view-skin.reducer.ts [hunk 13]
```diff
@@ -21,23 +23,27 @@ export interface AboutPartialState {
...
     steps: [
         {text: '1', data:  1, selected: true},
         {text: '10', data:  10, selected: false},
         {text: '50', data:  50, selected: false},
         {text: '100', data:  100, selected: false},
         {text: '200', data:  200, selected: false}
-    ]
+    ],
+    isShift: false,
+    isCtrl: false
```
Comment (sislex) [inline projects/tripod/src/lib/+state/skins/move-skin/view/move-view-skin.reducer.ts:39]: isCtrl: false,

File: projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts [hunk 14]
```diff
@@ -29,20 +29,23 @@ import {SkinMoveKeyboardEventsService} from '../../services/moveSkin/keyboardEve
...
+  private isLongPress: boolean = false;
 
   constructor(
     private readonly store: Store<MoveViewSkinState>,
     private skinMoveKeyboardEventsService: SkinMoveKeyboardEventsService,
     ) {}
 
-  handleKeyboardEvent(event: KeyboardEvent) {
-    this.skinMoveKeyboardEventsService.events(event);
+  handleKeyboardEvent(event: KeyboardEvent, note: any) {
```
Comment (sislex) [inline projects/tripod/src/lib/containers/control-buttons-container/control-buttons-container.ts:44]: note: string

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 15]
```diff
@@ -1,20 +1,120 @@
...
+        m: 2
+      }));
+      console.log('НАЖАЛИ l')
+    } else if (message.key === 'ArrowLeft' && note === 'up') {
+      this.isArrowLeftDown = false;
+      this.store.dispatch(sendDirection({
+        direction: 'LEFT',
+        m: 2
+      }));
+      console.log('ОТПУСТИЛИ l')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:60]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 16]
```diff
@@ -1,20 +1,120 @@
...
+      }));
+    }
+
+    else if (message.key === 'ArrowLeft' && note === 'down' && !this.isArrowLeftDown) {
+      this.isArrowLeftDown = true;
+      this.store.dispatch(sendDirection({
+        direction: 'LEFT',
+        m: 2
+      }));
+      console.log('НАЖАЛИ l')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:53]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 17]
```diff
@@ -1,20 +1,120 @@
...
+      console.log('ОТПУСТИЛИ l')
+    }
+    
+    else if (message.key === 'ArrowRight' && note === 'down' && !this.isArrowRightDown) {
+      this.isArrowRightDown = true;
+      this.store.dispatch(sendDirection({
+        direction: 'RIGHT',
+        m: 2
+      }));
+      console.log('НАЖАЛИ r')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:69]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 18]
```diff
@@ -1,20 +1,120 @@
...
+    } else if (message.key === 'ArrowRight' && note === 'up') {
+      this.isArrowRightDown = false;
+      this.store.dispatch(sendDirection({
+        direction: 'RIGHT',
+        m: 2
+      }));
+      console.log('ОТПУСТИЛИ r')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:76]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 19]
```diff
@@ -1,20 +1,120 @@
...
+        m: 2
+      }));
+      console.log('НАЖАЛИ u')
+    } else if (message.key === 'ArrowUp' && note === 'up') {
+      this.isArrowUpDown = false;
+      this.store.dispatch(sendDirection({
+        direction: 'UP',
+        m: 2
+      }));
+      console.log('ОТПУСТИЛИ u')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:92]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 20]
```diff
@@ -1,20 +1,120 @@
...
+    }
+
+    else if (message.key === 'ArrowDown' && note === 'down' && !this.isArrowDownDown) {
+      this.isArrowDownDown = true;
+      this.store.dispatch(sendDirection({
+        direction: 'DOWN',
+        m: 2
+      }));
+      console.log('НАЖАЛИ d')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:101]: remove console

File: projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts [hunk 21]
```diff
@@ -1,20 +1,120 @@
...
+    } else if (message.key === 'ArrowDown' && note === 'up') {
+      this.isArrowDownDown = false;
+      this.store.dispatch(sendDirection({
+        direction: 'DOWN',
+        m: 2
+      }));
+      console.log('ОТПУСТИЛИ d')
```
Comment (sislex) [inline projects/tripod/src/lib/services/moveSkin/keyboardEvents.service.ts:108]: remove console

**Pull request merged**